CalendarApp/Logic Handlers/CalendarEventHandler.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  CalendarEventHandler.swift
+//  CalendarApp
+//
+//  Created by Angelina Zhang on 7/18/22.
+//
+
+import Foundation
+import CalendarKit
+
+class CalendarEventHandler {
+    private var dateToCalendarKitEvents = [Date: [CalendarApp.Event]]()
+    private var calendar = Calendar(identifier: .gregorian)
+    
+    init() {
+        calendar.timeZone = TimeZone.current
+    }
+    
+    func getEventsForDate(_ date: Date) -> [CalendarApp.Event]? {
+        let midnight = calendar.startOfDay(for: date)
+        return dateToCalendarKitEvents[midnight]
+    }
+    
+    func addEvent(_ event: CalendarApp.Event, _ date: Date) {
+        let midnight = calendar.startOfDay(for: date)
+        var calendarEvents = dateToCalendarKitEvents[midnight] ?? [CalendarApp.Event]()
+        calendarEvents.append(event)
+        dateToCalendarKitEvents[midnight] = calendarEvents
+    }
+    
+    func addEventsFromArray(_ events:[CalendarApp.Event], _ date: Date) {
+        let midnight = calendar.startOfDay(for: date)
+        var calendarEvents = dateToCalendarKitEvents[midnight] ?? [CalendarApp.Event]()
+        for eventModel in events {
```

Comment:
nit, when you need to operate on everything in a collection it's better to use for each. This could be rewritten as: `events.forEach { calendarEvents.append($0) }`